controllers/kyma_controller.go

```diff
@@ -147,12 +145,32 @@ func (r *KymaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		if err := r.syncRemoteKymaSpecAndStatus(ctx, kyma); err != nil {
 			return r.CtrlErr(ctx, kyma, fmt.Errorf("could not synchronize remote kyma: %w", err))
 		}
+		if err := r.syncCrdsAndUpdateKymaAnnotations(ctx, kyma); err != nil {
```

Comment:
this condition is wrong, you still did update() but did not immediately quit the reconciling, if update success, there is no error returned, you will still continue to state handling, 

you should do something like this, meaning when you detected it needs to update, then update and return immediately. 
```
if syncCrdsAndUpdateKymaAnnotations() {
    if err := r.Update(ctx, kyma); err != nil {
			return err
    }
    return ctrl.Result{}, nil
}
```